### PR TITLE
Cache the newline split pattern in SourcePrinter for performance reasons

### PR DESCRIPTION
**Pull request by arturbosch**
More information on java not caching patterns: https://stackoverflow.com/questions/13420321/does-pattern-compile-cache

### CODE REVIEW

File: javaparser-core/src/main/java/com/github/javaparser/printer/SourcePrinter.java [hunk 1]
```diff
@@ -78,14 +79,19 @@ public SourcePrinter println() {
...
-    
+
     private StringBuilder bufAppend(final String arg) {
         updateCursor(arg);
         return buf.append(arg);
     }
 
+    private static final Pattern SPLIT_NEWLINE_PATTERN;
+    static {
+        SPLIT_NEWLINE_PATTERN = Pattern.compile("\r\n|\r|\n");
```
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/printer/SourcePrinter.java]: Egh, end of line characters defined somewhere in the code. Will have to look into that later.

Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/printer/SourcePrinter.java]: Won't a field initializer do instead of a static block?

Comment (arturbosch) [reply javaparser-core/src/main/java/com/github/javaparser/printer/SourcePrinter.java]: A static block does only initialize the field if this classes is loaded. If the user does not use it, it won't be loaded at all.

Comment (matozoid) [reply javaparser-core/src/main/java/com/github/javaparser/printer/SourcePrinter.java]: How does the performance gain compare to the loss of readability?

Comment (arturbosch) [reply javaparser-core/src/main/java/com/github/javaparser/printer/SourcePrinter.java]: Your right, the static block does do nothing in this case as this is not a static class and the pattern is always used.

Comment (arturbosch) [reply javaparser-core/src/main/java/com/github/javaparser/printer/SourcePrinter.java]: I haven't found such string somewhere in the code base. Tell me if you find something :)

**Comment by matozoid** [comment]
Hmmm, if I look at the code it seems like there is no point in all this end of line matching. All the sourceprinter ever receives is commands to print a piece of text without end of lines, and only the println method ever insert an EOL. @tarilabs - can you explain the reasoning behind this?

**Comment by tarilabs** [comment]
The change suggested by @arturbosch is correct; and sorry I missed that in the first place :(

@matozoid IIRC there is no check that the passed argument does not contain a newline. Additionally, please consider for instance when multi-line comments is supplied to the printer, it would "miss" some new lines if I remove this check (unless I overlooked something).

**Comment by matozoid** [comment]
@tarilabs ahhh, you're right, comments can interfere, and they can have different kinds of EOL characters. Good thing I'm refactoring all that comment stuff :-) Thanks for the answer!

**Pull request merged**